app/src/main/kotlin/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -269,7 +271,7 @@ class MessagingManager @Inject constructor(
         const val THIS_SERVER_ID = "server_id"
     }
 
-    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+    private val ioScope: CoroutineScope = CoroutineScope(Dispatchers.IO + Job())
```

Comment:
I'm not sure about this change. Also the `handleMessage` is so big ... I actually wonder if it would make sense to just make the whole thing running directly within the mainScope

Reply:
@jpelgrom do you have any idea here?

Reply:
> I'm not sure about this change. Also the `handleMessage` is so big ... I actually wonder if it would make sense to just make the whole thing running directly within the mainScope

I think that makes sense, this method has so much going on that its hard to escape from. its really the only way we can control the phone from the server

Reply:
What do I need to have an idea about? Making it a coroutine function sounds like a nice idea. Some of these actions may need to be running on the UI/main thread, like activity launches, so whatever we do we should test it.